### PR TITLE
Devo v2 dev

### PR DESCRIPTION
**Pull request by namrata-metron**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues

-  Refactored code for clarity and efficiency
-  Added error-checking mechanisms for robustness
-  Optimized database queries for faster execution
-  Improved documentation for ease of understanding and usage

## Description
This pull request introduces optimizations and improvements to the write_to_table_command and write_to_lookup_table_command functionalities. The changes enhance performance, improve error handling, and streamline code for better maintainability.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by namrata-metron — COMMENTED** [review]
enhancement changes

File: Packs/Devo/Integrations/Devo_v2/Devo_v2.py [hunk 1]
```diff
@@ -823,13 +824,42 @@ def multi_table_query_command(offset, items):
...
+        return json.dumps(value)
+    return str(value)
+
+
 def write_to_table_command():
-    table_name = demisto.args()["tableName"]
-    records = check_type(demisto.args()["records"], list)
+    tag = demisto.args().get("tag", None)
+    tableName = demisto.args()["tableName"]
+    records_str = demisto.args()["records"]
```
Comment (jbabazadeh) [inline Packs/Devo/Integrations/Devo_v2/Devo_v2.py:842]: You can use argToList function cause this records argument IsArray base on the .yml file
this function converts a string representation of args to a python list

and then you can remove the parsing and the check if the records its a list

File: Packs/Devo/ReleaseNotes/1_3_1.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#### Integrations
+
+##### Devo v2
+- Updated the Docker image to: *demisto/devo:1.0.0.87277*.
+
+- **Release Note Update:**
+  - Improved functionality for writing records to Devo tables and lookup tables has been updated.
+  - The Docker image has been updated to version *demisto/devo:1.0.0.86778*.
+
+These updates enhance the capabilities of the Devo integration by introducing improved methods for writing records to both Devo tables and lookup tables. The new functions provide enhanced flexibility and efficiency in managing and sending data to Devo's platform. Additionally, users can benefit from the latest version of the Docker image, ensuring compatibility and stability with the integration.
```
Comment (jbabazadeh) [inline Packs/Devo/ReleaseNotes/1_3_1.md:10]:
```suggestion
 - Improved functionality for **devo-write-to-table** and **devo-write-to-lookup-table** commands.
 - Updated the Docker image to: *demisto/devo:1.0.0.87277*.
```

**Review by jbabazadeh — CHANGES_REQUESTED** [review]
Hi @namrata-metron 
Thank you for your contribution!
Good work :)
Please see my comments.
Please feel free to reach out to me with any questions
Thanks again

File: Packs/Devo/ReleaseNotes/1_3_1.md (hunk 2 above)
Comment (jbabazadeh) [reply Packs/Devo/ReleaseNotes/1_3_1.md:10]: also here need to update the docker image- demisto/devo:1.0.0.87725

File: Packs/Devo/Integrations/Devo_v2/Devo_v2.py (hunk 1 above)
Comment (namrata-metron) [reply Packs/Devo/Integrations/Devo_v2/Devo_v2.py:842]: Actually the records list array consists of every datatypes i.e. list, boolean, integer, etc. So were directly fetching the data from records parameter. Else if we implement your suggestion, the XSOAR platform gives an error like this, "Failed to execute command devo-write-to-table. Error: the JSON object must be str, bytes or bytearray, not list."

File: Packs/Devo/ReleaseNotes/1_3_1.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#### Integrations
+
+##### Devo v2
+ - Improved functionality for **devo-write-to-table** and **devo-write-to-lookup-table** commands.
+ - Updated the Docker image to: *demisto/devo:1.0.0.87725*.
+
+These updates enhance the capabilities of the Devo integration by introducing improved methods for writing records to both Devo tables and lookup tables. The new functions provide enhanced flexibility and efficiency in managing and sending data to Devo's platform. Additionally, users can benefit from the latest version of the Docker image, ensuring compatibility and stability with the integration.
```
Comment (jbabazadeh) [inline Packs/Devo/ReleaseNotes/1_3_1.md:7]: Kindly exclude this as it does not in our standard release note format.
```suggestion
```

**Review by jbabazadeh — CHANGES_REQUESTED** [review]
@namrata-metron ,thanks for the changes you made.
 Please see my comment 
 Please add another test cause its still not reach the coverage is needed.
 Also you need to sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=32989)

File: Packs/Devo/Integrations/Devo_v2/Devo_v2.py (hunk 1 above)
Comment (jbabazadeh) [reply Packs/Devo/Integrations/Devo_v2/Devo_v2.py:842]: You need to add the records without enclosing brackets ([]),  as a list of strings separated by commas.

**Comment by namrata-metron** [comment]
Screenshot from 2024-02-22 11-28-01
This error is encountered when we try to execute `demisto-sdk pre-commit` command, whereas it isn't related to our code. So please guide us regarding the solution for the same.

**Comment by jbabazadeh** [comment]
Hi @namrata-metron try to revert the last changes after this commit [0ab5039](https://github.com/demisto/content/pull/32989/commits/0ab503948da14125424a4897b5776637da557320)

Try to run demisto-sdk pre-commit -g

**Comment by namrata-metron** [comment]
> Hi @namrata-metron try to revert the last changes after this commit [0ab5039](https://github.com/demisto/content/pull/32989/commits/0ab503948da14125424a4897b5776637da557320)
> 
> Try to run demisto-sdk pre-commit -g

Yes I have reverted the code and also added additional test cases as well.

**Comment by jbabazadeh** [comment]
Hi @namrata-metron ,

We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.  

Also can you please pull the changes from master? (merge from master into your branch)

**Comment by namrata-metron** [comment]
Hi @jbabazadeh,
We are available for the demo meeting today, and our time zone is IST.

**Review by namrata-metron — COMMENTED** [review]
Reviewed the changes

**Comment by jbabazadeh** [comment]
Hi @namrata-metron , 
Please let me know if you're free tomorrow, Tuesday, at 14:00 IST? 
and  share your email, I'll send you an invitation for the call.

**Comment by namrata-metron** [comment]
> Hi @namrata-metron , Please let me know if you're free tomorrow, Tuesday, at 14:00 IST? and share your email, I'll send you an invitation for the call.

Thanks for the update! And I'm available tomorrow for the demo meeting. Here is my mail Id: namrata.gachchi@metronlabs.com

**Comment by namrata-metron** [comment]
Hi @jbabazadeh,
Can you please confirm regarding the status of the PR, is it ready for the merge?

**Review by jbabazadeh — APPROVED** [review]
Thanks for the demo

**Comment by jbabazadeh** [comment]
Hi @namrata-metron ,
I merged the internal PR, and your update is now in the master branch of content repo.
Your update should be available in the marketplace within the next 72 hours.
Again, thank you very much for your contribution!

**Pull request merged**